core/src/main/java/org/geysermc/geyser/translator/sound/block/BucketSoundInteractionTranslator.java

```diff
@@ -38,7 +38,7 @@ public class BucketSoundInteractionTranslator implements BlockSoundInteractionTr
 
     @Override
     public void translate(GeyserSession session, Vector3f position, String identifier) {
-        if (session.getBucketScheduledFuture() == null) {
+        if (!session.isPlacedBucket()) {
```

Comment:
This is fine for now, but I don't think we need this in the future if we abstract out block/item handling properly like we discussed earlier.